data/notebooks/ingredients.py

```diff
@@ -619,39 +647,44 @@ def change_id(change):
     set_field(w_animal_group1, i.get("animal_group1"), None)
     set_field(w_animal_group2, i.get("animal_group2"), None)
     set_field(w_animal_product, i.get("animal_product"), None)
-    set_field(w_cropGroup, i.get("crop_group"), None)
-    set_field(w_landFootprint, i.get("land_occupation"), 0)
+    set_field(w_crop_group, i.get("crop_group"), None)
+    set_field(w_land_footprint, i.get("land_occupation"), 0)
 
 
 w_id.observe(change_id, names="value")
 
 
-def changed_database_to(field):
-    def changed_database(change):
-        results = list(dbsearch(change.new, w_search.value, limit=10))
-        field.rows = len(results)
-        field.options = [display_of(r) for r in results]
-        if results:
-            activity = results[0]
-            field.value = display_of(activity)
-            setattr(surfacebutton, "activity", activity)
-
-    return changed_database
-
-
 def changed_search_to(field):
     def changed_search(change):
-        results = list(dbsearch(w_database.value, change.new, limit=10))
+        activity = None
+        results = list(dbsearch(w_database.value, w_search.value, limit=10))
+        if w_search.value in [a["name"] for a in results]:
+            exacts = [a for a in results if a["name"] == w_search.value]
+            results = exacts
+            activity = results[0]
+        else:
+            if len(results) == 1:
+                activity = results[0]
+            else:
+                savebutton.disabled = True
+                savebutton.tooltip = savebuttontooltipnonunique
         field.rows = len(results)
         field.options = [display_of(r) for r in results]
-        if results:
-            activity = results[0]
+        if activity is not None:
             field.value = display_of(activity)
             setattr(surfacebutton, "activity", activity)
+            savebutton.disabled = False
+            savebutton.tooltip = savebuttontooltip
+        elif hasattr(surfacebutton, "activity"):
+            field.value = None
+            delattr(surfacebutton, "activity")
 
     return changed_search
 
 
+changed_database_to = changed_search_to
```

Comment:
Why not use `changed_search_to` directly?

Reply:
I just wanted to remind that changing the database in the UI might lead to a different behaviour than changing the search field